From 5773150c7e02df5ddb903ea058f9127493f4222f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Joan=20Bruguera=20Mic=C3=B3?= <joanbrugueram@gmail.com>
Date: Sat, 2 May 2026 12:58:31 +0000
Subject: [PATCH] Backport updated CMake versions to 4.05.04

This is a cherry-pick of upstream commits:
- cb51da4f5f23 ("Updated ACSVM") - partial
- a70b224edf3b ("Update cmake_minimum_version to silence all the warnings")
---
 acsvm/CMakeLists.txt         | 2 +-
 eecrashreport/CMakeLists.txt | 2 +-
 libpng/CMakeLists.txt        | 6 ++++--
 snes_spc/CMakeLists.txt      | 2 +-
 zlib/CMakeLists.txt          | 2 +-
 5 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/acsvm/CMakeLists.txt b/acsvm/CMakeLists.txt
index a6a3a654d..ac7f7d98f 100644
--- a/acsvm/CMakeLists.txt
+++ b/acsvm/CMakeLists.txt
@@ -10,7 +10,7 @@
 ##
 ##-----------------------------------------------------------------------------
 
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.6...3.31)
 
 if(COMMAND cmake_policy)
     cmake_policy(SET CMP0048 NEW)
diff --git a/eecrashreport/CMakeLists.txt b/eecrashreport/CMakeLists.txt
index 09262bff6..6e8535a3f 100644
--- a/eecrashreport/CMakeLists.txt
+++ b/eecrashreport/CMakeLists.txt
@@ -17,7 +17,7 @@
 ##
 ################################################################################
 ######################### CMake Configuration ##################################
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.10)
 if(COMMAND cmake_policy)
    cmake_policy(SET CMP0003 NEW)
 endif()
diff --git a/libpng/CMakeLists.txt b/libpng/CMakeLists.txt
index 86e0affc4..c34b7fe8c 100644
--- a/libpng/CMakeLists.txt
+++ b/libpng/CMakeLists.txt
@@ -25,8 +25,10 @@
 # For conditions of distribution and use, see the disclaimer
 # and license in png.h
 
-cmake_minimum_required(VERSION 3.1)
-cmake_policy(VERSION 3.1)
+# [DH] Bumped this from 3.1 to 3.5. Hopefully that will be OK.
+# IOANCH: Bumped to 3.10 due to deprecation warnings
+cmake_minimum_required(VERSION 3.10)
+cmake_policy(VERSION 3.10)
 
 project(libpng C ASM)
 enable_testing()
diff --git a/snes_spc/CMakeLists.txt b/snes_spc/CMakeLists.txt
index c9c47fc94..2c52cb30e 100644
--- a/snes_spc/CMakeLists.txt
+++ b/snes_spc/CMakeLists.txt
@@ -17,7 +17,7 @@
 ##
 ################################################################################
 ######################### CMake Configuration ##################################
-CMAKE_MINIMUM_REQUIRED (VERSION 2.4)
+CMAKE_MINIMUM_REQUIRED (VERSION 3.10)
 IF (COMMAND cmake_policy)
         CMAKE_POLICY (SET CMP0003 NEW)
 ENDIF (COMMAND cmake_policy)
diff --git a/zlib/CMakeLists.txt b/zlib/CMakeLists.txt
index 5675e264c..235ea1bd6 100644
--- a/zlib/CMakeLists.txt
+++ b/zlib/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.4.4)
+cmake_minimum_required(VERSION 3.10)
 set(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS ON)
 
 if(COMMAND cmake_policy)
-- 
2.54.0

